### PR TITLE
Fix color in firefox 2

### LINKED ISSUE #308: [Bug] Colour glitches on start since last update
https://github.com/user-attachments/assets/ca5b7884-319d-43d3-b119-48fadbc4cc1a

It only happens when I have the page open

https://github.com/user-attachments/assets/2683677c-d0e2-433b-8f68-b894ed49c51f

It also happens with other colours

### PR DESCRIPTION
**Pull request by giwih**
## Description
Fixed color glitches on start in FireFox, now colors dynamically change as in chrome, also in FireFox did not open [Test live](https://xengshi.github.io/materialYouNewTab/) because of bookmarks (now in unsupported browsers will be an error).

## Related Issues
- Issue Closes #308 

## Checklist
- [x] I have read and followed the [Contributing Guidelines](../CONTRIBUTING.md).
- [x] I have added necessary tests.
- [x] The project builds and runs without issues.

### CODE REVIEW

**Comment by itz-rj-here** [comment]
@prem-k-r

**Comment by Thunder-Blaze** [comment]
@XengShi Yes this should probably fix the issue, I'll check in a min

The ternary if statement is causing the main problem

File: script.js [hunk 1]
```diff
@@ -267,9 +267,15 @@ const bookmarkSearchClearButton = document.getElementById('clearSearchButton');
...
 const isFirefox = typeof browser !== 'undefined';
-var bookmarksAPI = isFirefox ? browser.bookmarks : chrome.bookmarks
+var bookmarksAPI;
+if (typeof browser !== 'undefined' && browser.bookmarks) {
+    bookmarksAPI = browser.bookmarks;
+} else if (typeof chrome !== 'undefined' && chrome.bookmarks) {
+    bookmarksAPI = chrome.bookmarks;
+} else {
+    console.error('Bookmarks API not supported in this browser.');
+}
```
Comment (prem-k-r) [inline script.js:278]: line 3460, please add condition there too, and move firefox prior to chrome.. as following
```
let bookmarksPermission;
        if (isFirefox && browser.permissions) {
            bookmarksPermission = browser.permissions;
        } else if (isChrome || isEdge || isBrave && chrome.permissions) {
            bookmarksPermission = chrome.permissions;
```

Comment (prem-k-r) [reply script.js:278]: @giwih

Comment (prem-k-r) [reply script.js:278]: we can add an alert here in else statement too

**Comment by prem-k-r** [comment]
@giwih conflicts :(

File: script.js [hunk 2]
```diff
@@ -3451,12 +3457,12 @@ document.addEventListener("DOMContentLoaded", function () {
 
     bookmarksCheckbox.addEventListener("change", function () {
         let bookmarksPermission;
-        if (isChrome || isEdge || isBrave) {
-            bookmarksPermission = chrome.permissions;
-        } else if (isFirefox) {
+        if (isFirefox && browser.permissions && isDesktop) {
```
Comment (prem-k-r) [inline script.js:3460]: Thanks :)

File: script.js [hunk 3]
```diff
@@ -269,12 +269,12 @@ const bookmarkViewList = document.getElementById('bookmarkViewList');
 
 const isFirefox = typeof browser !== 'undefined';
 var bookmarksAPI;
-if (typeof browser !== 'undefined' && browser.bookmarks) {
+if (isFirefox && browser.permissions && isDesktop) {
```
Comment (prem-k-r) [inline script.js:272]: ah, I didn't mean here too.. 
```
if (isFirefox && browser.bookmarks) {
```
is fine

File: script.js [hunk 4]
```diff
@@ -3451,12 +3457,12 @@ document.addEventListener("DOMContentLoaded", function () {
...
-        if (isChrome || isEdge || isBrave) {
-            bookmarksPermission = chrome.permissions;
-        } else if (isFirefox) {
+        if (isFirefox && browser.permissions && isDesktop) {
             bookmarksPermission = browser.permissions;
+        } else if (isChrome || isEdge || isBrave && chrome.permissions && isDesktop) {
+            bookmarksPermission = chrome.permissions;
         } else {
-            console.error("Unsupported Browser.");
+            alert("Bookmarks are not supported in your browser.");
```
Comment (prem-k-r) [inline script.js:3465]:
```
alert(translations[currentLanguage]?.UnsupportedBrowser || translations['en'].UnsupportedBrowser);
```

Comment (prem-k-r) [reply script.js:3465]:
```
"UnsupportedBrowser": "Bookmarks are not supported in your browser"
```

**Pull request merged**